src/pt/sussyscan/src/eu/kanade/tachiyomi/extension/pt/sussyscan/SussyScan.kt

```diff
@@ -1,34 +1,226 @@
 package eu.kanade.tachiyomi.extension.pt.sussyscan
 
-import eu.kanade.tachiyomi.multisrc.madara.Madara
+import android.annotation.SuppressLint
+import eu.kanade.tachiyomi.network.GET
 import eu.kanade.tachiyomi.network.interceptor.rateLimit
-import org.jsoup.nodes.Document
-import org.jsoup.nodes.Element
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.MangasPage
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.online.HttpSource
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.decodeFromStream
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.Interceptor
+import okhttp3.Request
+import okhttp3.Response
+import rx.Observable
+import uy.kohesive.injekt.injectLazy
 import java.text.SimpleDateFormat
-import java.util.Locale
-
-class SussyScan : Madara(
-    "Sussy Scan",
-    "https://oldi.sussytoons.com",
-    "pt-BR",
-    SimpleDateFormat("MMMM dd, yyyy", Locale("pt", "BR")),
-) {
-    override val client = super.client.newBuilder()
-        .rateLimit(2)
+import java.util.concurrent.TimeUnit
+
+class SussyScan : HttpSource() {
+
+    override val name = "Sussy Scan"
+
+    override val baseUrl = "https://new.sussytoons.site"
+
+    private val apiUrl = "https://api-dev.sussytoons.site"
+
+    override val lang = "pt-BR"
+
+    override val supportsLatest = true
+
+    // Moved from Madara
+    override val versionId = 2
+
+    private val json: Json by injectLazy()
+
+    override val client = network.cloudflareClient.newBuilder()
+        .rateLimit(1, 2, TimeUnit.SECONDS)
+        .addInterceptor(::imageLocation)
         .build()
 
-    override val useLoadMoreRequest = LoadMoreStrategy.Never
-    override val useNewChapterEndpoint = true
+    override fun headersBuilder() = super.headersBuilder()
+        .set("scan-id", "1")
+
+    // ============================= Popular ==================================
+
+    override fun popularMangaRequest(page: Int): Request {
+        val url = apiUrl.toHttpUrl().newBuilder()
+            .addPathSegments("obras/top5")
+            .build()
+        return GET(url, headers)
```

Comment:
There are no arguments passed to the HttpUrl builder, so it will always produce the same output.
```suggestion
        return GET("$apiUrl/obras/top5", headers)
```